e2e-tests/cypress/support/page-objects/invoices.ts

```diff
@@ -0,0 +1,177 @@
+    customerName: `Customer ${ id }`,
+    customerEmail: `customer${ id }@example.com`,
+    status: ['Paid', 'Unpaid', 'Pending'][Math.floor(Math.random() * 3)],
+    description: `Invoice for service ${ id }`,
+    address: `${ id } Main St, City, Country`,
+    phoneNumber: `1-555-${ id.toString().padStart(4, '0') }`,
+    taxId: `${ id.toString().padStart(3, '0') }-45-6789`,
+    paymentMethod: ['Credit Card', 'Bank Transfer', 'PayPal', 'Cash'][Math.floor(Math.random() * 4)],
+    notes: ['Paid in full', 'Awaiting payment', 'Pending approval'][Math.floor(Math.random() * 3)]
+  };
+}
+
+function generateInvoices(count: number): object[] {
+  const invoices: object[] = [];
+  for (let i = 1; i <= count; i++) {
+    invoices.push({create: {}});
+    invoices.push(generateInvoice(i));
+  }
+  return invoices;
+}
+
+function chunkArray(array: any[], chunkSize: number): any[][] {
+  const chunks = [];
+  for (let i = 0; i < array.length; i += chunkSize) {
+    chunks.push(array.slice(i, i + chunkSize));
+  }
+  return chunks;
+}
+
+function handleBulkResponse(response: string | undefined): void {
+  if (response === undefined) {
+    console.error('Response is undefined');
+    return;
+  }
+
+  try {
+    const items = response.items;
+
+    items.forEach((item: any) => {
+      const createResult = item.create;
+      if (createResult.status === 201) {
+        console.log(`Invoice ${createResult._id} created successfully.`);
+      } else {
+        console.error(`Failed to create invoice ${createResult._id}: ${createResult.error.reason}`);
+      }
+    });
+  } catch (error) {
+    console.error('Failed to parse response:', error);
+  }
+}
+
+function calculateSizeInBytes(object: any): number {
+  return Buffer.byteLength(JSON.stringify(object), 'utf8');
+}
+
+function checkBulkSizeLimit(invoices: object[], chunkSize: number): void {
+  const chunks = chunkArray(invoices, chunkSize);
+  const maxSizeInBytes = 100 * 1024 * 1024; // 100 MB
+
+  chunks.forEach((chunk, index) => {
+    const chunkSizeInBytes = calculateSizeInBytes(chunk);
+    if (chunkSizeInBytes > maxSizeInBytes) {
+      console.error(`Chunk ${ index + 1 } exceeds the 100 MB limit with size: ${ chunkSizeInBytes } bytes`);
+    } else {
+      console.log(`Chunk ${ index + 1 } is within the limit with size: ${ chunkSizeInBytes } bytes`);
+    }
+  });
+}
+
+function checkIndexExists(indexName: string): Cypress.Chainable<boolean> {
+  return cy.esGet({
+    endpoint: indexName,
+    credentials: 'kibana:kibana'
+
+  }).then((response: any) => {
+    return response.status !== 404;
+  });
+}
+
+function createIndex(indexName: string): Cypress.Chainable<void> {
+  return cy.esPut({
+    endpoint: indexName,
+    credentials: 'kibana:kibana',
+    printLogs: true
+  }).then((response: any) => {
+    console.log('Response:', response);
+    if (response) {
+      if (response.error && response.error.type === 'resource_already_exists_exception') {
```

Comment:
there is no need to do this. We should just assert that the response returned 2xx.